Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +352,873 @@ public override bool IsServerConnected()
+            test.SetValue("test_type", new EntityList<BaseEntity>()
+            {
+                data = new List<BaseEntity>()
+                {
+                    new BaseEntity()
+                    {
+                        TypeName = "list_node",
+                        Id = "list_node.test_type.regression"
+                    }
+                }
+            });
+
+            //foreach (ExternalItemFieldBase field in testCaseFields)
+            //{
+            //    if ((field.ToUpdate || field.Mandatory) && !test.Contains(field.ExternalID))
+            //    {
+            //        if (string.IsNullOrEmpty(field.SelectedValue) == false && field.SelectedValue != "NA")
+            //            test.SetValue(field.ExternalID, field.SelectedValue);                    
+            //    }
+            //}
+
+            test = Task.Run(() => { return this.octaneRepository.CreateEntity(GetLoginDTO(), test, null); }).Result;
+
+            activitiesGroup.ExternalID = test.Id.ToString();
+            activitiesGroup.ExternalID2 = test.Id.ToString();
+            CreateTestStep(test.Id, testScript);
+
+            return test.Id.ToString();
+        }
+
+        private string UpdateTestCase( ActivitiesGroup activitiesGroup, string fatherId, ObservableList<ExternalItemFieldBase> testCaseFields, string testScript)
+        {
+            TestManual test = new TestManual();
+            test.Id = activitiesGroup.ExternalID;
+            test.Name = activitiesGroup.Name;
+            test.SetValue("description", activitiesGroup.Description);
+            test.SetValue("automation_status", new BaseEntity()
+            {
+                TypeName = "list_node",
+                Id = "list_node.automation_status.not_automated"
+            });
+
+            test.SetValue("product_areas", new EntityList<BaseEntity>()
+            {
+                data = new List<BaseEntity>()
+                {
+                    new BaseEntity() {Id = fatherId, TypeName = "product_area"}
+                }
+            });
+            test.SetValue("test_type", new EntityList<BaseEntity>()
+            {
+                data = new List<BaseEntity>()
+                {
+                    new BaseEntity()
+                    {
+                        TypeName = "list_node",
+                        Id = "list_node.test_type.regression"
+                    }
+                }
+            });
+
+            test = Task.Run(() => { return this.octaneRepository.UpdateEntity(GetLoginDTO(), test, null); }).Result;
+
+            activitiesGroup.ExternalID = test.Id.ToString();
+            activitiesGroup.ExternalID2 = test.Id.ToString();
+            CreateTestStep(test.Id, testScript);
+
+            return test.Id.ToString();
+        }
+
+        private string CreateTestStep(string tcId, string script)
+        {
+            TestScript testScript = new TestScript();
+            testScript.Id = new EntityId(tcId.ToString());
```

Comment:
Codacy found an issue: [There's no need to call 'ToString()' on a string.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)